File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/utils/VerityUtil.java
package com.evernym.verity.sdk.utils;

import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.HttpClientBuilder;
import org.apache.http.util.EntityUtils;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

class VerityUtil {
    private VerityUtil(){}

    static Did retrieveVerityPublicDid(HttpGet request, HttpClient httpClient) throws IOException {
        HttpResponse response = httpClient.execute(request);
        int statusCode = response.getStatusLine().getStatusCode();
        if(statusCode > 399) {
            throw new IOException("Request failed! - " + statusCode + " - " + EntityUtils.toString(response.getEntity()));
        }
        else {
            String msg = EntityUtils.toString(response.getEntity());
            try {
                JSONObject msgObj = new JSONObject(msg);
                String did = msgObj.getString("DID");
                String verkey = msgObj.getString("verKey");
                return new Did(did, verkey);
            }
            catch (JSONException e){
                throw new IOException("Invalid and unexpected data from Verity -- response -- " + msg);
            }
        }
    }

    static Did retrieveVerityPublicDid(String verityUrl) throws IOException {
        HttpClient httpClient = HttpClientBuilder.create().build();

        String fullUrl = verityUrl+"/agency";
        HttpGet request = new HttpGet(fullUrl);

        return retrieveVerityPublicDid(request, httpClient);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Utils/VerityUtil.cs
using System;
using System.IO;
using System.Net.Http;
using System.Json;

namespace VeritySDK.Utils
{
    /// <summary>
    /// Utilities for basic interactions with the verity-application
    /// </summary>
    public class VerityUtil
    {
        public static Did retrieveVerityPublicDid(string verityUrl, HttpClient httpClient = null)
        {
            HttpClient client = httpClient ?? new HttpClient();

            string fullUrl = verityUrl + "/agency";
            var response = client.GetAsync(fullUrl).GetAwaiter().GetResult();

            var responseString = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();

            var statusCode = (int)response.StatusCode;
            if (statusCode > 399)
            {
                throw new IOException("Request failed! - " + statusCode + " - " + responseString);
            }
            else
            {
                try
                {
                    var msgObj = JsonObject.Parse(responseString);
                    return new Did(msgObj["DID"], msgObj["verKey"]);
                }
                catch (Exception e)
                {
                    throw new IOException("Invalid and unexpected data from Verity\r\nResponse -- " + responseString + "\r\nError -- " + e.Message);
                }
            }
        }
    }
}